docs/rules/id-length.md

```diff
@@ -0,0 +1,77 @@
+# Limit minimum and maximum length for identifiers (id-length)
+
+Identifier names such as `e`, `x`, `_t` or sometimes very long ones like `hashGeneratorResultOutputContainerObject` which usually makes the code harder to read and less maintainable. To prevent this, some code bases may enforce a minimum and/or maximum identifier length. (usually min 2-chars)
+
+```js
+// id-length: 1  // minimum 2-chars
+var x = 5; // too short
+```
+
+## Rule Details
+
+This rule is aimed at increasing code readability and maintainability by enforcing an identifier length convention. It will warn on any type of identifier which doesn't conform to length limits (upper and lower). The rule ignores common prefixes and suffixes up to two chars of `_` and `$` in any combination.
+
+It has exceptions for commonly used single-char identifier names such as `i` or `$`. It also allows the programmers to silently by-pass this check by using "quoted" property names or calculated property access to allow potential server-side data requirements.
+
+The following patterns are considered warnings:
+
+```js
+// id-length: 1
+var x = 5;
+function _f() { return 42; }
+obj.$e = document.body;
+var handler = function (e) { /* do stuff */ };
+try {
+    dangerousStuff();
+} catch (e) {  // Identifier 'e' is too short. (< 2)
+    // ignore as many do
+}
+var myObj = { a: 1 };
+```
+
+The following patterns are not considered warnings:
+
+```js
+// id-length: 1
+var num = 5;
+function _func() { return 42; }
+obj.$el = document.body;
+var handler = function (evt) { /* do stuff */ };
+try {
+    dangerousStuff();
+} catch (error) {  // Identifier 'e' is too short. (< 2)
+    // ignore as many do
+}
+var myObj = { apple: 1 };
+var data = { "x": 1 };  // excused because of quotes
+data["y"] = 3;  // excused because of calculated property access
+```
+
+### Options
+
+The `id-length` rule has no required options and has 4 optional ones that needs to be passed in a single options object:
+
+* **min** *(default: 2)*: The minimum number of characters an identifier name should be, after it is stripped from it is prefixes and suffixes
+* **max** *(default: Infinity)*: The maximum number of characters an identifier name should be, after it is stripped from it is prefixes and suffixes
+* **exceptions** *(default: ["$", "_", "i", "j"])*: An array of identifier names that the rule should not apply to
+* **disableFor** *(default: [])*: An array of node types that the rule should not be applied to. Available types are:
```

Comment:
Is this necessary? I'm generally not a big fan of requiring node types in configuration and I don't see a clear use case for this.